pkg/proxy/downstream.go

```diff
@@ -1217,12 +1218,20 @@ func (s *downStream) onUpstreamHeaders(endStream bool) {
 }
 
 func (s *downStream) handleUpstreamStatusCode() {
-	// todo: support config?
 	if s.upstreamRequest != nil && s.upstreamRequest.host != nil {
-		if s.requestInfo.ResponseCode() >= http.InternalServerError {
+		responseCode := s.requestInfo.ResponseCode()
+
+		switch {
+		case responseCode >= http.BadRequest && responseCode < http.InternalServerError:
+			s.upstreamRequest.host.HostStats().UpstreamResponseClientError.Inc(1)
+			s.upstreamRequest.host.ClusterInfo().Stats().UpstreamResponseClientError.Inc(1)
```

Comment:
missing `UpstreamResponseSuccess` here?

this looks a bit complicated now.

Reply:
I will fix that

Reply:
It's hard to decide, unless we give these keys a whole new meaning: `UpstreamResponseSuccess` and `UpstreamResponseFailed` represent L4 status, and HTTP status codes represent L7 status (like envoy https://www.envoyproxy.io/docs/ envoy/latest/configuration/http/http_conn_man/stats#statistics)

Reply:
oh, I just mean the backward compatibility.